editor/src/components/canvas/design-panel-root.tsx

```diff
@@ -49,6 +49,20 @@ function isCodeEditorEnabled(): boolean {
 
 const DesignPanelRootInner = React.memo(() => {
   const roomStatus = useStatus()
+
+  const room = useRoom()
+  const loginStateType = useEditorState(
+    Substores.userState,
+    (store) => store.userState.loginState.type,
+    'DesignPanelRootInner loginStateType',
+  )
+
+  React.useEffect(() => {
+    if (loginStateType === 'LOGGED_IN' && roomStatus !== 'connected') {
```

Comment:
I think this should check that `roomStatus === 'disconnected'` instead, so you don't have potential concurrent reconnect tries.

Reply:
added it in https://github.com/concrete-utopia/utopia/pull/4654/commits/f155c775b738b2713131172106335dead7225f70